cohorts/load.py

```diff
@@ -1020,6 +1020,93 @@ def plot_joint(self, on, on_two=None, **kwargs):
         p = sb.jointplot(data=df, x=plot_cols[0], y=plot_cols[1])
         return p
 
+    def summarize_provenance(self):
+        """ Utility function to summarize provenance of Cohort data for each existing cache_dir
+            
+            == Returns == 
+
+            Dict containing unique provenance among all patients in each existing cache dir, or 
+            None if provenance is not unique among all patients.
+
+            Dict also contains an item 'dfhash' representing unique hash of default dataframe for cohort, given options.
+            
+            == Behavior == 
+
+            Checks for non-equivalent provenance among patients in each cache dir that exists. Cache dirs that
+            do not exist are not populated in the dictionary.
+
+            If non-equivalence is found:
+               - prints a warning
+               - returns None for that cache dir in the resulting dict
+
+            Otherwise:
+               - returns unique provenance for that cache dir, equivalent among all patients in data frame
+            
+            == See also ==
+            
+            `?cohorts.summarize_unique_provenance` which optionally prints a single provenance dict if provenance is identical across all
+            cache_dirs.
+
+        """
+        provenance_summary = {}
+        df = self.as_dataframe()
+        df_hash = hash(str(df.sort_values('patient_id')))
+        for cache in self.cache_names:
+            cache_name = self.cache_names[cache]
+            cache_provenance = None
+            cache_warnings = ""
```

Comment:
These get printed by `_compare_provenance()`. This object is just used to assess its length. Should _compare_provenance instead return a true/false?